python/semantic_kernel/kernel.py

```diff
@@ -222,7 +227,7 @@ async def invoke_prompt(
             raise TemplateSyntaxError("The prompt is either null or empty.")
 
         function = KernelFunctionFromPrompt(
-            function_name=function_name,
+            function_name=function_name or generate_random_ascii_name(),
```

Comment:
Thanks for handling this here. Could you add `Closes #6804` to your PR description, please?